src/client/a2/main/grid/layers/pixi-effects.component.ts

```diff
@@ -0,0 +1,92 @@
+import * as _ from 'lodash';
+import { Component, Input, OnInit, AfterViewInit } from '@angular/core';
+
+import { RadarEffect, RadarEffectFactory } from './effects/radar-effect-factory';
+
+@Component({
+  selector: 'pixi-effects',
+  templateUrl: 'pixi-effects.component.pug',
+  styleUrls: ['pixi-effects.component.styl'],
+})
+
+export class PixiEffectsComponent implements OnInit, AfterViewInit {
+  private renderer: PIXI.SystemRenderer;
+  private container: PIXI.Container;
+  private width: number;
+  private height: number;
+
+  private radarEffectFactory: RadarEffectFactory;
+  private radarEffects: RadarEffect[];
+
+  constructor() {
+    this.radarEffects = [];
+    this.radarEffectFactory = new RadarEffectFactory();
+  }
+
+  ngOnInit() {
+    
+  }
+
+  ngAfterViewInit() {   
+    let element = document.getElementById('canvas');
+    //let element = document.getElementsByClassName('canvas')[0];
+    this.width = element.clientWidth;
+    this.height = element.clientHeight;
+
+    console.log(this.width);
+    console.log(this.height);
+
+    this.initRenderer();
+
+    element.appendChild(this.renderer.view);
+
+    this.render();
+  }
+
+  private initRenderer() {
+    this.renderer = PIXI.autoDetectRenderer(this.width, this.height, { transparent: true });
+    this.renderer.autoResize = true;
+    this.container = new PIXI.Container();
+  }
+
+  createRadarEffect(x: number, y: number, size: number, speed: number) {
+    let effect = this.radarEffectFactory.create(0.05, [1.0, 0.33, 0.13, 1.0], size, speed);
+    this.radarEffects.push(effect);
+    effect.setPosition(x, y);
+
+    this.container.addChild(effect.getDisplayObject());
+  }
+
+  render() {
+    let radarEffects = [];
+    _.forEach(this.radarEffects, function(effect) {
+      effect.update();
+      if (false === effect.isActive()) {
+        effect.getDisplayObject().destroy();
+      } else {
+        radarEffects.push(effect);
```

Comment:
Is there a better way to remove an item while looping through the list? I thought these are just "pointer\reference" assignment rather than copy by values (is there a way to do reference assignment)?